client_test.go

```diff
@@ -312,6 +311,85 @@ func TestClientFast(t *testing.T) {
 	CheckNumMessages(t, rabbitC, queueName, 0)
 }
 
+func TestPanicHandlingInConsumer(t *testing.T) {
+	t.Parallel()
+	rabbitC, url := runRabbitContainer(t)
+	defer func() { require.NoError(t, rabbitC.Terminate(context.Background())) }()
+
+	const dlqQueueName = "queue-dlq"
+	const dlqExchangeName = "exchange-dlq"
+	const queueName = "queue"
+	require.NoError(t, declareQueue(t, rabbitC, dlqQueueName))
+	require.NoError(t, declareExchange(t, rabbitC, dlqExchangeName, "fanout"))
+	require.NoError(t, declareBinding(t, rabbitC, dlqExchangeName, dlqQueueName))
+	require.NoError(t, declareQueueWithDlq(t, rabbitC, queueName, dlqExchangeName))
+
+	num := 10
+	received := make(chan bool, num) // this channel gets "released" on message delivery
+	// Consumer
+	qi, err := NewQueue(url, queueName, 1, true, false, nil)
+	require.NoError(t, err)
+	defer qi.Close()
+
+	const numberForWhichPublisherPanic = 4
+	rec := func(m amqp.Delivery) *amqp.Publishing {
+		if m.CorrelationId == strconv.Itoa(numberForWhichPublisherPanic) {
+			panic("panicking !!!!!")
+		}
+
+		assert.NoError(t, m.Ack(false))
+		received <- true
+		return nil
+	}
+	assert.NoError(t, qi.AddReceiver(rec))
+
+	// Publisher
+	jobOutChannel := make(chan amqp.Delivery)
+	qo, err := NewQueue(url, queueName, 1, false, false, jobOutChannel)
+	assert.NoError(t, err)
+	defer qo.Close()
+	pub := func(d amqp.Delivery) *amqp.Publishing {
+		return &amqp.Publishing{
+			ContentType:   d.ContentType,
+			CorrelationId: d.CorrelationId,
+			Body:          d.Body,
+			Headers:       d.Headers,
+		}
+	}
+	assert.NoError(t, qo.AddPublisher(context.TODO(), pub))
+
+	var wg sync.WaitGroup
+	wg.Add(num)
+	for i := 0; i < num; i++ {
+		i := i
+		go func() {
+			defer wg.Done()
+			jobOutChannel <- amqp.Delivery{CorrelationId: strconv.Itoa(i)}
+
+			if i != numberForWhichPublisherPanic {
+				<-received
+			}
+		}()
+	}
+	wg.Wait()
+
+	CheckNumMessages(t, rabbitC, queueName, 0)
+	CheckNumMessages(t, rabbitC, dlqQueueName, 1)
+
+	t.Run("check rejected message", func(t *testing.T) {
+		dlq, err := NewQueue(url, dlqQueueName, 1, false, false, nil)
+		require.NoError(t, err)
+		defer dlq.Close()
+
+		m, close, err := dlq.GetReadyMessages(1)
```

Comment:
is it possible to check it multiple messages ?

Reply:
`num` messages is sent in that test